File: Contratistas/contratistas-csv/src/main/java/com/contratistas/contratistascsv/servicios/ExportCSV.java
package com.contratistas.contratistascsv.servicios;

import com.contratistas.contratistascsv.model.CSVContratistas;
import org.springframework.stereotype.Service;
import java.io.BufferedWriter;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.List;

@Service
public class ExportCSV {

        public void export (List<CSVContratistas> filas, Path outputPath) throws IOException {
            if(filas.isEmpty()){
                throw new IllegalStateException("No hay filas validas para generar el archivo CSV");
            }
            if(outputPath.getParent()!= null){
                Files.createDirectories(outputPath.getParent());
            }
            try (BufferedWriter writer = Files.newBufferedWriter(outputPath)){
            writer.write("Nombre Completo, Telefono,Correo Electronico,Empresa,Ciudad,Correo Corportativo generado");
            writer.newLine();
            for(CSVContratistas fila : filas){
                    writer.write(csvfila(fila.Nombre())+ ","
                            + csvfila(fila.Telefono())+ ","
                            + csvfila(fila.Email())
                            + csvfila(fila.Empresa())+ ","
                            + csvfila(fila.Ciudad())+ ","
                            + csvfila(fila.Email_corporativo())+ ","

                )
            }

            }

        }
    private String csvfila(String valor){
            String safe = valor == null ? "": valor.trim();
            return "\"" + safe.replace("\"", "\"\"") + "\"";

    }
}

File: Contratistas/Python/app/services/exportCSV.py
import csv
from pathlib import Path
from app.config import settings

def export_csv(filas):
    path = Path(settings.output_csv)
    path.parent.mkdir(parents=True, exist_ok=True)

    with open(path, mode="w", newline="", encoding="utf-8") as file:
        writer = csv.writer(file)
        writer.writerow(["Nombre", "Telefono", "Correo", "Empresa", "Ciudad", "Correo Corporativo"])

        for fila in filas:
            writer.writerow([
                fila.nombre,
                fila.telefono,
                fila.correo,
                fila.empresa,
                fila.ciudad,
                fila.correo_corporativo
            ])